sympy/solvers/ode.py

```diff
@@ -1482,6 +1491,233 @@ def _test_term(coeff, order):
+        sol = (sol).subs(x, match['mobius'])
+        sol = sol.subs(x, x**match['k'])
+        e = (e).subs(x, x**match['k'])
+
+        if B:
+            e1 = Integral(A/2, x).doit()
+            e1 = exp(logcombine(e1, force=True))
+
+        sol = cancel(powdenest(cancel(e/e1)*x**((-match['k']+1)/2), force=True))*(sol)
+        sol = Eq(func, sol)
+
+    # if sol is None then we can try for series solution
+    if sol is None:
+        try:
+            sol = dsolve(eq, func, hint = "2nd_power_series_regular", x0=x0)
+        except ValueError:
+            pass
+    if sol is None:
+        raise NotImplementedError("....")
+
+    return sol
+
+def match_2nd_2F1_hypergeometric(I, k, sing_point, func, r):
+
+    from sympy import factor
+    x = func.args[0]
+    a = Wild("a")
+    b = Wild("b")
+    c = Wild("c")
+    t = Wild("t")
+    alpha = Wild("alpha")
+    beta = Wild("beta")
+    gamma = Wild("gamma")
+    delta = Wild("delta")
+
+    rn = {'type':None}
+    # I0 of the standerd 2F1 equation.
+    I0 = ((a-b+1)*(a-b-1)*x**2 + 2*((1-a-b)*c + 2*a*b)*x + c*(c-2))/(4*x**2*(x-1)**2)
+    if sing_point != [0, 1]:
+        # If singular point is [0, 1] then we have standerd equation.
+        eqs = []
+        sing_eqs = [-beta/alpha, -delta/gamma, (delta-beta)/(alpha-gamma)]
+        # making equations for the finding the mobius transformation
+        for i in range(3):
+            if i<len(sing_point):
+                eqs.append(Eq(sing_eqs[i], sing_point[i]))
+            else:
+                eqs.append(Eq(1/sing_eqs[i], 0))
+        # solving above equations for the mobius transformation
+        _beta = solve(eqs[0], beta)[0]
+        _delta = solve(eqs[1], delta)[0]
+        _gamma = solve(((eqs[2]).subs(beta, _beta)).subs(delta, _delta), gamma)[0]
+        subs = (alpha*x + beta)/(gamma*x + delta)
+        subs = subs.subs(beta, _beta)
+        subs = subs.subs(delta, _delta)
+        subs = subs.subs(gamma, _gamma)
+        subs = cancel(subs)
+    else:
+        subs = x
+    t = (solve(Eq(subs, t), x)[0]).subs(t, x)
+    # applying mobius transformation in I to make it into I0.
+    I = I.subs(x, t)
+    I = I*(t.diff(x))**2
+    I = factor(I)
+
+    dict_I = {x**2:0, x:0, 1:0}
+    I0_num, I0_dem = I0.as_numer_denom()
+    # collecting coeff of (x**2, x), of the standerd equation.
+    dict_I0 = {x**2:(a-b)**2 - 1, x:2*((1-a-b)*c + 2*a*b), 1:c*(c-2)}
+    # collecting coeff of (x**2, x) from I0 of the given equation.
+    dict_I.update(collect(expand(cancel(I*I0_dem)), [x**2, x], evaluate=False))
+    eqs = []
+    # We are comparing the coeff of powers of different x, for finding the values of
+    # parameters of standerd equation.
+    for key in [x**2, x, 1]:
+        eqs.append(Eq(dict_I[key], dict_I0[key]))
+
+    _c = solve(eqs[2], c)[0]
+    t = solve(eqs[0].subs(a-b, t), t)[0]
+    _a = t+b
+    _b = solve((eqs[1].subs(c, _c)).subs(a, _a), b)[0]
```

Comment:
If these are quadratics it's probably better to solve using `roots` than `solve`.

Reply:
okay, I will try.